files/usr/share/cinnamon/applets/recent@cinnamon.org/applet.js

```diff
@@ -87,7 +100,12 @@ MyApplet.prototype = {
     },
 
     _display: function() {
-        this._refreshRecents();
+        if (global.settings.get_boolean(PANEL_EDIT_MODE_KEY)) {
+            this._refreshRecents();
+            this._on_panel_edit_mode_changed();
+        } else {
+            this._on_panel_edit_mode_changed();
+        }
```

Comment:
This can just be:
```
this._refreshRecents();
this._on_panel_edit_mode_changed();
```
The panel-edit-mode key is already being checked in this._on_panel_edit_mode_changed(), we don't need to do it here also.